src/components/ContactDetailsModal/ContactDetailsModal.js

```diff
@@ -0,0 +1,263 @@
+import AddressScanner from 'components/QRCodeScanner/AddressScanner';
+import Icon from 'components/Icon';
+
+// utils
+import { fontStyles, spacing } from 'utils/variables';
+import { images } from 'utils/images';
+import { getThemeColors, themedColors } from 'utils/themes';
+import { isEnsName, isValidAddress } from 'utils/validators';
+import { addressesEqual } from 'utils/assets';
+
+import { lookupAddress } from 'utils/common';
+import { getReceiverWithEnsName } from 'utils/contacts';
+
+// types
+import type { Theme } from 'models/Theme';
+import type { Contact } from 'models/Contact';
+
+
+type Props = {
+  theme: Theme,
+  isVisible: boolean,
+  onModalHide: () => void,
+  onSavePress: (contact: Contact) => void,
+  contact: ?Contact,
+  dirtyInputs?: boolean,
+  isDefaultNameEns?: boolean,
+  title?: string,
+  contacts: Contact[],
+  showQRScanner?: boolean,
+};
+
+const InputWrapper = styled.View`
+  flex-direction: row;
+  align-items: center;
+  margin-top: ${spacing.small}px;
+`;
+
+const FieldIcon = styled(CachedImage)`
+  width: 48px;
+  height: 48px;
+  margin-right: ${spacing.small}px;
+`;
+
+const ErrorMessage = styled(BaseText)`
+  margin-top: ${spacing.large}px;
+  text-align: center;
+  color: ${themedColors.negative};
+  ${fontStyles.small};
+`;
+
+export const LoadingSpinner = styled(Spinner)`
+  margin-top: ${spacing.large}px;
+  align-items: center;
+  justify-content: center;
+`;
+
+export const TitleWrapper = styled.View`
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+`;
+
+export const QRCodeButton = styled.TouchableOpacity`
+  position: absolute;
+  top: 0;
+  right: 0;
+`;
+
+
+export const QRCodeIcon = styled(Icon)`
+  color: ${({ color }) => color};
+  font-size: 20px;
+`;
+
+const renderContactInput = (
+  value: string,
+  onChangeText: (value: string) => void,
+  placeholder: string,
+  icon: string,
+  theme: Theme,
+) => (
+  <InputWrapper>
+    <FieldIcon source={icon} />
+    <TextInput
+      theme={theme}
+      inputWrapperStyle={{ flex: 1, paddingBottom: 0 }}
+      inputProps={{
+        value,
+        onChangeText,
+        placeholder,
+      }}
+    />
+  </InputWrapper>
+);
+
+const ContactDetailsModal = ({
+  isVisible,
+  onModalHide,
+  theme,
+  contact,
+  onSavePress,
+  isDefaultNameEns,
+  title,
+  contacts,
+  showQRScanner,
+}: Props) => {
+  const [addressValue, setAddressValue] = useState('');
+  const [nameValue, setNameValue] = useState('');
+  const [dirtyInputs, setDirtyInputs] = useState(false);
+  const [resolvingEns, setResolvingEns] = useState(false);
+  const [isScannerVisible, setIsScannerVisible] = useState(false);
+  const { walletIcon, personIcon } = images(theme);
+
+  // reset input value on default change
+  useEffect(() => {
+    setDirtyInputs(false);
+    setAddressValue(contact?.ethAddress || '');
+    const defaultName = isDefaultNameEns ? contact?.ensName : contact?.name;
+    setNameValue(defaultName || '');
```

Comment:
It's more like stating that the value here is string type. It can definitely go as null tho, but then more edge cases needs to be covered to support nullish value rather than just string. Should I change it?

Reply:
Resolving this as this appears to be UI related.